content/docs/10.plugin-developer-guide/07.outputs.md

```diff
@@ -2,40 +2,98 @@
+Kestra can track outputs and metrics from any scripting language. The core [Python](../../plugins/core/tasks/scripts/io.kestra.core.tasks.scripts.Python.md) and [Node](../../plugins/core/tasks/scripts/io.kestra.core.tasks.scripts.Node.md) plugins provide methods to configure outputs and metrics.
 
-If you are using another language like [Bash](../../plugins/core/tasks/scripts/io.kestra.core.tasks.scripts.Bash.md), we don't provide any package to help you, but you can do it with the `echo` command on stdout.
+If you use another language such as [Bash](../../plugins/core/tasks/scripts/io.kestra.core.tasks.scripts.Bash.md), you can leverage the `echo` command on the `stdout`.
 
 ## Script command
 
-Kestra looks at every outputs on standard out (or standard err) and search for the special patterns `::{}::` or `{}` that allow to specify outputs and metrics as JSON objects:
-- `{}` can be used for single line JSON objects.
+Kestra inspects every output on standard output and standard error and looks for the special patterns `::{}::` or `{}` that allow to specify outputs and metrics as JSON objects:
+- `{}` can be used for single-line JSON objects.
 - `::{}::` can be used for multi-line JSON objects.
 
-Here is the representation of the output object, it's a map in which you can add any key/value pair:
+Below is an example of an output object. It's a **dictionary** object that you can use to add any key-value pairs:
 
-```json5
-"outputs": { // map of key value with all the outputs you want to send
+```json
+"outputs": { // map/dicttionary of key-value pairs with outputs
     "my-key": "my-value",
-    "my-list": [1, 2, 3] // you can use  json type (bool, array, map, …)
+    "my-list": [1, 2, 3] // you can use  a variety of data types supported in a JSON request payload
 }
 ```
 
-Here is the representation of the metrics object, it's a list of JSON objects:
-```json5
-"metrics": [ // you can send multiple metrics at once
+Here is the representation of a `metrics` object. It's a **list of dictionaries**:
+```json
+"metrics": [ // you can send multiple metrics at once - each is a dictionary within a list
     {
         "name": "my-counter", // mandatory, the name of the metrics
-        "type": "counter", // mandatory, "counter" or "timer"
-        "value": 1.2, // mandatory (double), counter to add, or duration in seconds for timer
-        "tags":{ // optional list of tags that will expose internal details
-          "type":"read",
-          "location":"eu"
+        "type": "counter", // mandatory, "counter" or "timer" metric type
+        "value": 1.2, // mandatory (double), counter to measure number of objects/rows/items processed in a given task, or duration in seconds for the timer metric
+        "tags": { // optional list of tags that will expose internal details
+          "type": "read",
+          "location": "EU"
         }
     }
 ]
 ```
 
+## When to use metrics and when to use outputs?
+
+### Use cases for outputs
```

Comment:
outputs are also used a lot to pass data between tasks

Reply:
yup, I'll make that more explicit